crates/doc/src/ptr.rs

```diff
@@ -236,7 +245,7 @@ impl Pointer {
             // which we'll create the next child location.
             if let HeapNode::Null = v {
                 match token {
-                    Token::Property(_) => {
+                    Token::Property(_) | Token::AdditionalProperties => {
```

Comment:
i think the `NextIndex` and `NextProperty` cases should just return `None` here -- these locations cannot be created.